### PR TITLE
[ci] Obtain Lychee with Bazel (@lychee//:lychee)

### PR DESCRIPTION
**Pull request by dmcardle**
This commit makes it possible to run ci/scripts/check-links.sh without worrying which version of Lychee is installed locally.

### CODE REVIEW

File: third_party/lychee/repos.bzl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:repo.bzl", "http_archive_or_local")
+
+def lychee_repos(local = None):
+    LYCHEE_VERSION="v0.11.1"
```
Comment (dmcardle) [inline third_party/lychee/repos.bzl:8]: @HU90m Looks like v0.12.0 came out two weeks ago on <https://github.com/lycheeverse/lychee/releases>. Should we upgrade to that?

**Comment by dmcardle** [comment]
@HU90m Curious what you think of this change. I wanted to play with the CI script locally, but quickly realized I didn't have Lychee installed. I think we might as well track the dependency in Bazel so we never have to worry which version is installed on which machines.

**Comment by dmcardle** [comment]
I've been having a hard time getting CI to pass on this PR, probably due to the CI slowdown other folks are seeing. I just experimentally rebased on top of 8186cbb9b8a87e8d297ecb2c72a8e0c93e4cdd3c after seeing @a-will's comment that it's the CI performance baseline. If CI now passes in the usual amount of time, then it seems likely that a commit between 8186cbb9b8a87e8d297ecb2c72a8e0c93e4cdd3c...b0a9c9c3a1fa97ad1e3c0201f7604cf595269e9d is responsible for the CI slowdown.

**Comment by a-will** [comment]
> I've been having a hard time getting CI to pass on this PR, probably due to the CI slowdown other folks are seeing. I just experimentally rebased on top of the older commit [8186cbb](https://github.com/lowRISC/opentitan/commit/8186cbb9b8a87e8d297ecb2c72a8e0c93e4cdd3c) after seeing @a-will's comment that it's the CI performance baseline. If CI now passes in the usual amount of time, then it seems likely that a commit between [8186cbb...b0a9c9c](https://github.com/lowRISC/opentitan/compare/8186cbb9b8a87e8d297ecb2c72a8e0c93e4cdd3c...b0a9c9c3a1fa97ad1e3c0201f7604cf595269e9d) is responsible for the CI slowdown. (This PR was originally based on [b0a9c9c](https://github.com/lowRISC/opentitan/commit/b0a9c9c3a1fa97ad1e3c0201f7604cf595269e9d).)

Heh, to do that, you'd have to modify the CI flow. CI does its own merge with master, and it brought you up to this merge commit: https://github.com/lowRISC/opentitan/commit/5cdb866a9c44fb622d355f31f75f3cd079b2dde9

This appears to be how `steps.checkout` works in Azure Pipelines w/ GitHub. You'd probably have to do something manually to the repo to override.

**Comment by dmcardle** [comment]
> Heh, to do that, you'd have to modify the CI flow. CI does its own merge with master, and it brought you up to this merge commit: [5cdb866](https://github.com/lowRISC/opentitan/commit/5cdb866a9c44fb622d355f31f75f3cd079b2dde9)
> 
> This appears to be how `steps.checkout` works in Azure Pipelines w/ GitHub. You'd probably have to do something manually to the repo to override.
> 
> Alternatively, you could do a git revert of the range of commits you want to remove.

Whoops, I didn't realize this is how CI works. Thanks for explaining that, @a-will.

Just rebased on tip-of-tree this morning.

**Review by HU90m — APPROVED** [review]
Yeah, this is much nicer, thanks Dan

**Comment by dmcardle** [comment]
Thanks, @HU90m! Just added a trivial commit that bumps the Lychee dep to v0.12.0.

**Review by milesdai — APPROVED** [review]
LGTM!

**Comment by milesdai** [comment]
@dmcardle I think you also need to update the hash with the version bump

**Pull request merged**